components/nexusoperations/config.go

```diff
@@ -68,6 +70,32 @@ ScheduleNexusOperation commands with an operation name that exceeds this limit w
 Uses Go's len() function to determine the length.`,
 )
 
+var MaxOperationHeaderSize = dynamicconfig.NewNamespaceIntSetting(
+	"component.nexusoperations.limit.header.size",
+	4096,
+	`The maximum allowed header size for a Nexus Operation.
+ScheduleNexusOperation commands with a "nexus_header" field that exceeds this limit will be rejected.
+Uses Go's len() function on header keys and values to determine the total size.`,
+)
+
+var DisallowedOperationHeaders = dynamicconfig.NewNamespaceTypedSettingWithConverter(
+	"component.nexusoperations.disallowedHeaders",
+	func(a any) ([]string, error) {
+		keys, ok := a.([]string)
+		if !ok {
+			return nil, fmt.Errorf("expected a string slice, got: %v", a)
+		}
+		for i, k := range keys {
+			keys[i] = strings.ToLower(k)
+		}
+		return keys, nil
+	},
+	[]string{"request-timeout"},
```

Comment:
Thinking we may also want to disallow headers we use internally:
* https://github.com/temporalio/temporal/blob/main/common/headers/headers.go#L47
* https://github.com/temporalio/temporal/blob/main/common/rpc/interceptor/redirection.go#L50

Reply:
Good point, let me do that.

Reply:
Also did you consider allowing patterns? I'm thinking in the future it might be nice to have something like disallow all `x-temporal-*` headers. But that might be overkill at the moment so I'm fine with leaving it as is for now.

Reply:
Done.

Reply:
Hmm... yeah we could consider that but lets start out with this.